lib/server.js

```diff
@@ -100,24 +100,61 @@ function parsePOST(req) {
     }
 }
 
+function logResponse(opts, response) {
+    var logLevel = 'trace/request';
+    if (response.status >= 500) {
+        logLevel = 'error/request';
+    } else if (response.status >= 400) {
+        logLevel = 'info/request';
+    }
+    opts.log(logLevel, {
+        message: response.message,
+        res: response,
+        stack: response.stack
+    });
+}
+
+/**
+ * Removes Hop-To-Hop headers that might be
+ * proxied from a backend service
+ *
+ * @param {Object} rh response headers
+ * @private
+ */
+function removeHopToHopHeaders(rh) {
+    // Need to delete properties and not set to undefined
+    // because node passes 'undefined' to client.
```

Comment:
Since `delete` is quite expensive, it might be worth checking that those headers are actually present before deleting.